test/middleware/service.availability.middleware.spec.ts

```diff
@@ -109,4 +122,64 @@ describe("service availability middleware tests", () => {
 
     expect(response.text).not.toContain("Service offline - Apply to remove an overseas entity from the register");
   });
+
+  describe("Remove feature flag tests for remove specific pages", () => {
+    beforeEach(() => {
+      jest.resetAllMocks();
+    });
+
+    test(`Redirects to the next page whenever the remove flag is true`, async () => {
+      mockIsActiveFeature
+        .mockReturnValueOnce(false) // SHOW_SERVICE_OFFLINE_PAGE
+        .mockReturnValueOnce(true) // FEATURE_FLAG_ENABLE_ROE_UPDATE
+        .mockReturnValueOnce(true); // FEATURE_FLAG_ENABLE_ROE_REMOVE
+      const resp = await request(app).get(`${REMOVE_CANNOT_USE_URL}?${PREVIOUS_PAGE_QUERY_PARAM}=${REMOVE_SOLD_ALL_LAND_FILTER_PAGE}`);
+      expect(resp.status).toEqual(302);
+      expect(resp.header.location).toEqual(`/signin?return_to=${SECURE_UPDATE_FILTER_URL}`);
+    });
+
+    test(`Does not reach the ${REMOVE_CANNOT_USE_PAGE} when the remove flag is false`, async () => {
+      mockIsActiveFeature
+        .mockReturnValueOnce(false) // SHOW_SERVICE_OFFLINE_PAGE
+        .mockReturnValueOnce(true) // FEATURE_FLAG_ENABLE_ROE_UPDATE
+        .mockReturnValueOnce(false); // FEATURE_FLAG_ENABLE_ROE_REMOVE
+      const resp = await request(app).get(`${REMOVE_CANNOT_USE_URL}?${PREVIOUS_PAGE_QUERY_PARAM}=${REMOVE_SOLD_ALL_LAND_FILTER_PAGE}`);
+      expect(resp.status).toEqual(200);
+      expect(resp.text).not.toContain("You cannot apply to remove this overseas entity");
+      expect(resp.text).not.toContain("Service unavailable");
+    });
+
+    test(`Does not reach the ${REMOVE_SOLD_ALL_LAND_FILTER_PAGE} when the remove flag is false`, async () => {
+      mockIsActiveFeature
+        .mockReturnValueOnce(false) // SHOW_SERVICE_OFFLINE_PAGE
+        .mockReturnValueOnce(true) // FEATURE_FLAG_ENABLE_ROE_UPDATE
+        .mockReturnValueOnce(false); // FEATURE_FLAG_ENABLE_ROE_REMOVE
+      const resp = await request(app).get(`${REMOVE_SOLD_ALL_LAND_FILTER_URL}?${PREVIOUS_PAGE_QUERY_PARAM}=${REMOVE_SOLD_ALL_LAND_FILTER_PAGE}`);
+      expect(resp.status).toEqual(200);
+      expect(resp.text).not.toContain("You cannot apply to remove this overseas entity");
+      expect(resp.text).not.toContain("Service unavailable");
+    });
+
+    test(`Does not reach the ${REMOVE_IS_ENTITY_REGISTERED_OWNER_PAGE} when the remove flag is false`, async () => {
+      mockIsActiveFeature
+        .mockReturnValueOnce(false) // SHOW_SERVICE_OFFLINE_PAGE
+        .mockReturnValueOnce(true) // FEATURE_FLAG_ENABLE_ROE_UPDATE
+        .mockReturnValueOnce(false); // FEATURE_FLAG_ENABLE_ROE_REMOVE
+      const resp = await request(app).get(`${REMOVE_IS_ENTITY_REGISTERED_OWNER_URL}?${PREVIOUS_PAGE_QUERY_PARAM}=${REMOVE_SOLD_ALL_LAND_FILTER_PAGE}`);
+      expect(resp.status).toEqual(200);
+      expect(resp.text).not.toContain("You cannot apply to remove this overseas entity");
```

Comment:
the text "You cannot apply to remove this overseas entity" should only be on the REMOVE_CANNOT_USE_PAGE. Each of the 'Does not reach" tests would need .not.contain. text from the page you are requesting, eg REMOVE_IS_ENTITY_REGISTERED_OWNER_PAGE in this test.